models/repo/repo_list.go

```diff
@@ -730,3 +730,21 @@ func GetUserRepositories(opts *SearchRepoOptions) (RepositoryList, int64, error)
 	repos := make(RepositoryList, 0, opts.PageSize)
 	return repos, count, db.SetSessionPagination(sess, opts).Find(&repos)
 }
+
+func GetPrimaryRepoLanguageList(ctx context.Context) (LanguageStatList, error) {
+	languageList := make(LanguageStatList, 0)
```

Comment:
Since we just need `language`, we can use a string slice here directly.

Reply:
This is still an improvement.

Reply:
The Colour is also needed, not only the name